victron_ble/scanner.py

```diff
@@ -72,25 +73,42 @@ async def start(self):
         logger.info(f"Reading data for {list(self._device_keys.keys())}")
         await super().start()
 
-    def callback(self, device: BLEDevice, data: bytes):
-        advertisement_key = self._device_keys.get(device.address.lower())
-        if advertisement_key is None:
-            return
+    def get_device(self, ble_device: BLEDevice, raw_data: bytes) -> Device:
+        address = ble_device.address.lower()
+        if address not in self._known_devices:
+            advertisement_key = self.load_key(address)
 
-        if device.address not in self._known_devices:
-            device_klass = detect_device_type(data)
+            device_klass = detect_device_type(raw_data)
             if not device_klass:
-                logger.error(f"Could not identify device type for {device}")
-                return
+                raise UnknownDevice(f"Could not identify device type for {ble_device}")
+
+            self._known_devices[address] = device_klass(advertisement_key)
+        return self._known_devices[address]
+
+    def load_key(self, address: str) -> str:
+        try:
+            return self._device_keys[address]
+        except KeyError:
+            raise AdvertisementKeyMissing(f"No key available for {address}")
 
-            self._known_devices[device.address] = device_klass(advertisement_key)
+    def callback(self, ble_device: BLEDevice, raw_data: bytes):
+        logger.debug(
+            f"Received data from {ble_device.address.lower()}: {raw_data.hex()}"
+        )
+        try:
+            device = self.get_device(ble_device, raw_data)
+        except AdvertisementKeyMissing as e:
+            logger.debug(e)
```

Comment:
Could we skip this debug logging here? I think this introduces logging noise from unrelated Victron devices which I wanted to avoid in this particular Scanner implementation.

Reply:
Sure, done.

There'll still be one debug event logged a few lines up, but it won't look like an error.